src/Trace/Sampler/SampleState.php

```diff
@@ -30,4 +30,17 @@ public function __construct(
         $this->headers = $headers;
         $this->traceOptions = $traceOptions;
     }
+
+    public function __toString(): string
+    {
+        return sprintf(
+            'SampleState(decision=%d, attributes=%s, settings=%s, traceState=%s, headers=%s, traceOptions=%s)',
+            $this->decision,
+            implode(',', $this->attributes->toArray()),
+            $this->settings ?? 'null',
+            $this->traceState ?? 'null',
+            $this->headers,
```

Comment:
Is `headers` ok without a default?

Reply:
Yes, it is because the `__toString()` method handled the null case of the header. So it never returns null